content/docs/use-cases/ci-cd-for-machine-learning.md

```diff
@@ -0,0 +1,98 @@
+# Continuous Integration and Deployment for Machine Learning
+
+Applying DevOps methodologies to machine learning (MLOps) and data management
+(DataOps) is increasingly common. This means resource orchestration
+(provisioning servers for model training), model testing (validating model
+inference), and model deployment to production, as well as monitoring &
+feedback. [DVC](/) can manage data/models and reproducible pipelines, while
+[CML] can assist with orchestration, testing and monitoring.
+
+[cml]: https://cml.dev
+
+![](https://static.iterative.ai/img/ml-pipeline.png) _Basic ML pipeline_
+
+The main benefits of CI/CD (continuous integration and continuous delivery) for
+ML are detailed below.
+
+Firstly, we can automate and enforce testing. This includes enforcing integrity
+with application-specific tests:
+
+- **Data validation**: for example, validation against a schema or verifying
+  pipeline consistency -- correct shapes, data types, etc.
+- **Model validation**: for example, input/output and performance validation --
+  all dependencies present for inference to run, and model scores within
+  thresholds.
+
+Typically, this requires **infrastructure orchestration**: making CI systems
+provision and launch a GPU instance, train a model, cleanly terminate, and pull
+results back. In particular:
+
+- **Provisioning**: When data or code is added/changed in a pull request (PR),
+  automatically running validation tests in the cloud.
+- **Metrics**: Automatically generating metrics reports.
+
+Finally, this can be taken a step further with entire **experiments**/research
+being run online:
+
+- Fine-tune on a schedule: set up jobs to pull in new data from regularly
+  updated external sources to retrain/refine deployed models.
+- Run a hyperparameter search: keep heavyweight resources and big data on CI
+  servers running overnight, and put your laptop to sleep.
+
+Normally, CI/CD is hard to set up, configure, and maintain -- especially for
+data and ML pipelines.
+
+![](https://static.iterative.ai/img/ml-vs-cicd.png) _Traditional ML meets CI/CD_
+
+[DVC](/) and [CML] removes most of this
+[management headache](https://papers.nips.cc/paper/2015/file/86df7dcfd896fcaf2674f757a2463eba-Paper.pdf)
+in ML. You can automate all of the above without needing any additional
+configuration. Here are a few feature highlights:
+
+**Models, Data, and Metrics as Code**: DVC removes the need to create versioning
+databases, use special file/folder structures, or write bespoke interfacing
```

Comment:
This last part is about the solution right? Structure-wise I think it's OK to have a list here only if no bullet lists remain in the first half. And even then maybe it can be a list of brief (1/2 sentence) features.

In any case there would still need to be some more text that complete and conclude the story in some logical way. Probably elaborating on how this features work for the user's advantage. Again, no need to go super deep into details so hopefully the whole thing can be about 1/2 oe 2/3 the length of this to my mind, based on other use cases.